src/wasm/wasm-ir-builder.cpp

```diff
@@ -2483,11 +2483,16 @@ Result<> IRBuilder::makeStackSwitch(HeapType ct, Name tag) {
   }
   StackSwitch curr(wasm.allocator);
   curr.tag = tag;
-  auto nparams = ct.getContinuation().type.getSignature().params.size();
+  Type params = ct.getContinuation().type.getSignature().params;
+  auto nparams = params.size();
   if (nparams < 1) {
     return Err{"arity mismatch: the continuation argument must have, at least, "
                "unary arity"};
   }
+  if (!params[nparams - 1].isContinuation()) {
+    return Err{"the last argument of the continuation argument should be "
+               "itself a continuation"};
+  }
```

Comment:
Can we leave this to the validator? It doesn't look like `ChildTyper::visitStackSwitch` depends on the last parameter correctly being a continuation.

Reply:
Thanks for your time!
I have just removed this for now since validation is not implemented.